lib/ansible/modules/storage/netapp/netapp_e_lun_mapping.py

```diff
@@ -96,57 +76,24 @@
 import json
 
 from ansible.module_utils.api import basic_auth_argument_spec
-from ansible.module_utils.pycompat24 import get_exception
-from ansible.module_utils.urls import open_url
-
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.six.moves.urllib.error import HTTPError
+from ansible.module_utils.netapp import request, eseries_host_argument_spec
+from ansible.module_utils.pycompat24 import get_exception
 
 HEADERS = {
     "Content-Type": "application/json",
     "Accept": "application/json"
 }
 
 
-def request(url, data=None, headers=None, method='GET', use_proxy=True,
-            force=False, last_mod_time=None, timeout=10, validate_certs=True,
-            url_username=None, url_password=None, http_agent=None, force_basic_auth=True, ignore_errors=False):
-    try:
-        r = open_url(url=url, data=data, headers=headers, method=method, use_proxy=use_proxy,
-                     force=force, last_mod_time=last_mod_time, timeout=timeout, validate_certs=validate_certs,
-                     url_username=url_username, url_password=url_password, http_agent=http_agent,
-                     force_basic_auth=force_basic_auth)
-    except HTTPError:
-        err = get_exception()
-        r = err.fp
-
-    try:
-        raw_data = r.read()
-        if raw_data:
-            data = json.loads(raw_data)
-        else:
-            raw_data = None
-    except:
-        if ignore_errors:
-            pass
-        else:
-            raise Exception(raw_data)
-
-    resp_code = r.getcode()
-
-    if resp_code >= 400 and not ignore_errors:
-        raise Exception(resp_code, data)
-    else:
-        return resp_code, data
-
-
-def get_host_and_group_map(module, ssid, api_url, user, pwd):
+def get_host_and_group_map(module, ssid, api_url, user, pwd, validate_certs):
     mapping = dict(host=dict(), group=dict())
 
     hostgroups = 'storage-systems/%s/host-groups' % ssid
     groups_url = api_url + hostgroups
     try:
-        hg_rc, hg_data = request(groups_url, headers=HEADERS, url_username=user, url_password=pwd)
+        hg_rc, hg_data = request(groups_url, headers=HEADERS, url_username=user, url_password=pwd,
```

Comment:
Future improvement: Along the same line of thought as above, should we rename request to something like santricity_request?   Just trying to improve the readability.